Guide multi app solution (#1085)
Altinn/altinn-studio-docs: content/app/guides/multi-app-solution/instructions/app-a/_index.en.md

---
title: Application A
linktitle: Application A
description: Instructions for setting up application A
weight: 10
aliases:

- /app/multi-app-solution/instructions/app-a

---

> Application A is responsible for trigger the instantiation, and passing data to an instance of Application B. In order

Reviewer: ```suggestion
Application A is responsible for triggering the instantiation of Application B and passing data to it. 
```

> to achieve this some steps must be taken.

Reviewer: ```suggestion
 To accomplish this, several steps must be followed.
```

1. Extend the application process with an additional process task.
2. Add logic for instantiation of application B to an event trigger.

> 3. Populate/retrieve and pass on relevant data to an instance of application B.

Reviewer: ```suggestion
3. Populate/retrieve and pass on relevant data to the newly created instance of application B.
```

{{% notice warning %}}

> The following guidelines assumes that a basic Altinn application already exists so the

Reviewer: ```suggestion
These guidelines assume that a basic Altinn application already exists. The focus is on the
```

> more technical adaptations can be enhanced for realizing the purpose of the multi-app solution

Reviewer: ```suggestion
more technical adaptations necessary for realizing the purpose of the multi-app solution
```

{{% /notice %}}

## Add task to process

### Why the need for multiple tasks

In most cases it is necessary to pass the data the end user
has added to the form, further on to application B. This data is preserved in the pdf-element which is stored as a part
of the instance object. However, this pdf is only generated at the _end_ of a process task. Hence there is need for an
extra task to be able to retrieve the pdf. You will need to have at least two tasks, where the
final task is _not_ a data task. The task can be a _confirm_
or _feedback_ task type. Since we recommend using the _confirm_ task type, this is what the following guidelines will
use.

> _NB: It might be that you need to get an updated version of

Reviewer: Does this actually way anything about why we need multiple tasks or could we delete the sentence?
